sd-export/send-to-usb

```diff
@@ -0,0 +1,84 @@
+#! /usr/bin/python3
+
+import datetime
+import json
+import os
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+
+DEVICE = "/dev/sda1"
+MOUNTPOINT = "/media/usb"
+ENCRYPTED_DEVICE = "encrypted_volume"
+FILE = sys.argv[1]
+folder_name = ""
+encryption_method = ""
+encryption_key = ""
+target_folder = "sd-export-{}".format(datetime.datetime.now().isoformat())
+
+tmpdir = tempfile.mkdtemp()
+
+if os.path.exists(FILE):
+
+    try:
+        with tarfile.open(FILE) as tar:
+            tar.extractall(tmpdir)
+    except Exception as e:
+        # exit with 0 return code otherwise the os will attempt to open
+        # the file with another application
+        print("Error opening export bundle")
+        sys.exit(0)
+
+    try:
+        folder_name = os.path.basename(FILE).split(".")[0]
+        with open(os.path.join(tmpdir, folder_name, "metadata.json")) as json_data:
+            data = json.load(json_data)
+            encryption_method = data["encryption-method"]
+            encryption_key = data["encryption-key"]
+    except Exception as e:
+        print("Error parsing metadata")
+        sys.exit(0)
+
+    # we only support luks for now
+    if encryption_method != "luks":
+        print("Unsupported export encryption")
+        sys.exit(0)
+
+    # the luks device is not already unlocked
+
+    if not os.path.exists(os.path.join("/dev/mapper/", ENCRYPTED_DEVICE)):
+        p = subprocess.Popen(
+            ["sudo", "cryptsetup", "luksOpen", DEVICE, ENCRYPTED_DEVICE],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        stdout_data = p.communicate(input=str.encode(encryption_key, "utf-8"))[0]
+        rc = p.returncode
+        if rc != 0:
+            print("Bad passphrase or luks error")
+            sys.exit(0)
+
+    # mount target not created
+    if not os.path.exists(MOUNTPOINT):
+        out = subprocess.call(["sudo", "mkdir", MOUNTPOINT])
+        out = subprocess.call(["sudo", "chown", "-R", "user:user", MOUNTPOINT])
+    out = subprocess.call(
+        ["sudo", "mount", os.path.join("/dev/mapper/", ENCRYPTED_DEVICE), MOUNTPOINT]
+    )
```

Comment:
Nit: `subprocess.call` returns the exit code from the run command, not output from the command, as implied by "out". As with the subsequent `subprocess.call`s, fine to omit the `out = ` altogether, since we're not using it. 

Might be worth using `check_call` instead and catching CalledProcessError to exit 0, otherwise we can't know that the commands are succeeding. In general, error handling in all the mimetype handler scripts is tricky, as we don't want to fail over over to an alternative solution.